### ISSUE TITLE
is it possible to build win .exe's on linux?

### ISSUE THREAD

**Issue by mark-summerfield** [pwiecz/go-fltk#50]
For command line applications I can usually build them on linux for windows using a command line like this:
```env GOOS=windows GOARCH=amd64 go build -o myapp.exe .
````
But when I try this with a go-fltk app I get an error:
```
github.com/pwiecz/go-fltk: build constraints exclude all Go files in /home/mark/opt/gows/pkg/mod/github.com/pwiecz/go-fltk@v0.0.0-20221205050836-2ee6d591498f
```
It is possible, or must I build for windows _on_ windows?

**Comment by pwiecz** [comment]
I cannot read the command line.
Can you write it again?

**Comment by MoAlyousef** [comment]
github.com/pwiecz/go-fltk: build constraints exclude all Go files in /home/mark/opt/gows/pkg/mod/github.com/pwiecz/go-fltk@v0.0.0-20221205050836-2ee6d591498f

It seems related to this issue:

https://github.com/golang/go/issues/29074

**Comment by mark-summerfield** [comment]
The command line was:

env GOOS=windows GOARCH=amd64 go build -o myapp.exe .

**Comment by MoAlyousef** [comment]
I was able to get it working by specifying CGO_ENABLED=1 and specifying the cross-compiler, since according to the issue, the built-in C/C++ compiler in the golang toolchain isn't cross-compiling:
```
CGO_ENABLED=1 GOOS=windows GOARCH=amd64 CC=x86_64-w64-mingw32-gcc CXX=x86_64-w64-mingw32-g++ go build
```
(uses the mingw cross-compiler)

I was also able to get things working using the Zig toolchain:
```
CGO_ENABLED=1 GOOS=windows GOARCH=amd64 CXX="zig c++ -target x86_64-windows-gnu" CC="zig cc -target x86_64-windows-gnu" go build
```

**Comment by mark-summerfield** [comment]
I started by installing this: `go install github.com/tc-hib/go-winres@latest`. Then I installed the mingw cross-compiler. Then I changed my `buildwin.sh` script to:
```
go-winres simply --icon images/icon.ico
CGO_ENABLED=1 GOOS=windows GOARCH=amd64 CC=x86_64-w64-mingw32-gcc \
    CXX=x86_64-w64-mingw32-g++ \
    go build -buildvcs=false -ldflags -H=windowsgui -o myapp.exe .
```
Go insisted I add the -buildvcs=false flag and since I didn't want a console to pop up I added the -ldflags -H=windowsgui flag.

This created myapp.exe which won't run on win 7, but runs perfectly on win 10.

Thank you!
